site/docs/manual/Architecture.md

```diff
@@ -130,35 +173,111 @@ entity.addComponent(ComponentA)
 the engine will try to reuse a `ComponentA` instance, from the pool of components previously created, and it won't allocate a new one instead.
 When releasing that component, by calling `entity.removeComponent(ComponentA)`, it will get returned to the pool, ready to be used by another entity.
 
-ECSY should know how to reset a component to its original state, that's why it's highly recommended that components implements a `reset` method to get the benefits from pooling.
+ECSY should know how to reset a component to its original state, if your component has the proper schema defined, ECSY will do this for you.
+
+#### Custom Components
+
+Sometimes it's not possible to define a component with a schema. If you still want to get the benefits of object pooling you can override some of the methods on the `Component` class.
 
 ```javascript
-// Example of components with `reset` methods implemented
+class ColorArray extends Component {
```

Comment:
can't you just define the schema and avoid using contructor, but define your own copy/reset functions?

Reply:
Yeah, maybe I should be explicit that not all of these are necessary. I just wanted to cover all the things you can override.